homeassistant/components/smarttub/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""SmartTub integration."""
+import logging
+
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN, SMARTTUB_CONTROLLER
+from .controller import SmartTubController
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup(hass, _config):
+    """Set up smarttub component."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up a smarttub config entry."""
+
+    controller = SmartTubController(hass)
+    hass.data[DOMAIN][entry.unique_id] = {
+        SMARTTUB_CONTROLLER: controller,
+    }
+
+    if not controller.coordinator.last_update_success:
```

Comment:
Doesn't look like there's a coordinator before calling `controller.async_setup_entry`. Maybe best to raise if `controller.async_setup_entry` returns `False`.